pandas/core/groupby/ops.py

```diff
@@ -755,6 +759,17 @@ def get_iterator(
         """
         splitter = self._get_splitter(data, axis=axis)
         keys = self.group_keys_seq
+        if self.raise_warning_single_grouper:
```

Comment:
This method is called by many places internally where the behavior of pandas will not change from a user perspective, e.g. `DataFrameGroupBy._transform_general`. I think we only want to warn when a user is iterating over a groupby object themselves. If you think this is wrong, please let me know!

Reply:
@rhshadrach 
I made it in a way if the passed key is a single element in a list. I think if it's internally looping it shouldn't use a single list key so it won't raise a warning. This was my take.
How do you think I can do it in a way that guarantees to raise the error only if when the user themselves iterate over groupby

Reply:
I believe the user should be accountable for passing a single element without a list when using `DataFrameGroupBy._transform_general` or we can edit the function internally to fix the user's input as done in the plotting functions

Reply:
> How do you think I can do it in a way that guarantees to raise the error only if when the user themselves iterate over groupby

The `__iter__` method is called when iterating, e.g. `for index, group in df.groupby('a')`. I believe that's where this warning should go. This method may also be called internally by pandas; it would be okay to modify such calls to use `self.grouper.get_iterator...` instead so as to avoid warnings generated internally.

Reply:
@rhshadrach Can you see it now?